neuron/src/app/Neuron/CLI/Search.hs

```diff
@@ -11,27 +11,37 @@ module Neuron.CLI.Search
   )
 where
 
+import qualified Data.Text as Text
+import Development.Shake (Action)
 import Neuron.CLI.Rib
+  ( SearchBy (SearchByContent, SearchByTitle),
+    SearchCommand (..),
+  )
+import Neuron.Config.Type (Config, getZettelFormats)
+import Neuron.Reader.Type (ZettelFormat, zettelFormatToExtension)
 import Relude
 import System.Posix.Process
 import System.Which
 
 neuronSearchScript :: FilePath
 neuronSearchScript = $(staticWhich "neuron-search")
 
-searchScriptArgs :: SearchCommand -> [String]
-searchScriptArgs SearchCommand {..} =
-  let searchByArgs =
+searchScriptArgs :: (NonEmpty ZettelFormat) -> SearchCommand -> [String]
+searchScriptArgs formats SearchCommand {..} =
+  let extensionPattern = "/*{" <> (Text.unpack $ Text.intercalate "," $ toList $ zettelFormatToExtension <$> formats) <> "}"
+      searchByArgs =
         case searchBy of
-          SearchByTitle -> ["(^# )|(^title: )", "2"]
-          SearchByContent -> ["", "2"]
+          SearchByTitle -> ["(^# )|(^title: )|(^\\* )", "2", extensionPattern]
```

Comment:
I haven't found a good way to make a pattern specific to a correct extension. Any suggestions?

Reply:
Hmm, that's unfortunate. 

If no solution can be found, we should probably remove the `^\\*` matching entirely and just say in the documentation that `neuron search` (not full-text) is unsupported for .org files. Your PR would then effectively make it work for full-text search only.

Reply:
In that `neuron search` should fail right away, if org is enabled in neuron.dhall. But `neuron search --full-text` should work fine regardless. 

Unless you have got other suggestions ...

Reply:
Thanks for the pointer. I'll look into it.

Reply:
Updated. Can you review when having a chance?